src/mbgl/gl/attribute.hpp

```diff
@@ -242,7 +247,18 @@ class Attributes {
     static constexpr std::size_t Index = TypeIndex<A, As...>::value;
 
     static Locations bindLocations(const ProgramID& id) {
-        return Locations { bindAttributeLocation(id, Index<As>, As::name())... };
+        std::set<std::string> activeAttributes;
```

Comment:
You can minimize the declarations and code here in the header by writing:

```C++
std::set<std::string> activeAttributes = getActiveAttributes(id);
```

Then `getActiveAttributes` is the only thing that needs to be declared; everything else can go in the .cpp.